examples/appprotect-dos/README.md

```diff
@@ -0,0 +1,69 @@
+# NGINX App Protect Dos Support
+
+In this example we deploy the NGINX Plus Ingress controller with [NGINX App Protect Dos](https://www.nginx.com/products/nginx-app-protect-dos/), a simple web application and then configure load balancing and BADOS protection for that application using the Ingress resource.
+
+## Running the Example
+
+## 1. Deploy the Ingress Controller
+
+1. Follow the installation [instructions](../../docs/installation.md) to deploy the Ingress controller with NGINX App Protect Dos.
+
+2. Save the public IP address of the Ingress controller into a shell variable:
+    ```
+    $ IC_IP=XXX.YYY.ZZZ.III
+    ```
+3. Save the HTTPS port of the Ingress controller into a shell variable:
+    ```
+    $ IC_HTTPS_PORT=<port number>
+    ```
+
+## 2. Deploy the Webapp Application
+
+Create the webapp deployment and service:
+```
+$ kubectl create -f webapp.yaml
+```
+
+## 3. Configure Load Balancing
+1. Create the syslog service and pod for the App Protect Dos security logs:
+    ```
+    $ kubectl create -f syslog.yaml
+    ```
+2. Create a secret with an SSL certificate and a key:
+    ```
+    $ kubectl create -f webapp-secret.yaml
+    ```
+3. Create the App Protect Dos policy and log configuration:
+    ```
+    $ kubectl create -f apdos-policy.yaml
+    $ kubectl create -f apdos-logconf.yaml
+    ```
+4. Create an Ingress Resource:
+
+    Update the `appprotectdos.f5.com/app-protect-dos-security-log-destination` annotation from `webapp-ingress.yaml` with the ClusterIP of the syslog service. For example, if the IP is `10.101.21.110`:
+    ```yaml
+    . . .
+    appprotect.f5.com/app-protect-security-log-destination: "syslog:server=10.101.21.110:514"
```

Comment:
I think we've talked about this in another channel, but I think it's good to add it here too so it's not lost. 

at the moment this annotation uses the cluster IP of the service to write log files. This is fine for normal nginx, but in Kubernetes, the cluster IP can change between different clusters or on service remove/add. 

The suggested fix is that the annotation specifies the 'namespace/name:port' of the service instead and a lookup is performed to get the cluster IP.

something like:
```
svcObj, svcExists, err := lbc.svcLister.GetByKey(svcKey)
ip := svcObj.Spec.ClusterIP
```

Reply:
Just a further thought, if we have four Ingress controllers, will they all point at the same logging service? Will that service have multiple replicas ? i.e 4 ingresses and 4 loggers ? In which case how do we control which logs go where? How do we correlate which log is from which ingress?